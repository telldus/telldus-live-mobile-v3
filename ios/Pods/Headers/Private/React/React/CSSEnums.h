../../../../../../node_modules/react-native/React/CSSLayout/CSSEnums.h