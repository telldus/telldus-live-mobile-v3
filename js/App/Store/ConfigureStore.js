/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 */

'use strict';

import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import promise from './Promise';
import array from './Array';
import analytics from './Analytics';
import reducers from 'Reducers';
import createLogger from 'redux-logger';
import { persistStore, autoRehydrate } from 'redux-persist';
import { AsyncStorage } from 'react-native';

let isDebuggingInChrome = __DEV__ && !!window.navigator.userAgent;

let logger = createLogger({
	predicate: (getState, action) => isDebuggingInChrome,
	collapsed: true,
	duration: true,
});

let createTheStore = applyMiddleware(thunk, promise, array, analytics, logger)(createStore);

let _store;
export function configureStore(onComplete: ?() => void) {
	const store = autoRehydrate()(createTheStore)(reducers);
	persistStore(store, {storage: AsyncStorage}, onComplete);
	if (isDebuggingInChrome) {
		window.store = store;
	}
	_store = store; // TODO: fix this ugly stuff
	return store;
}
export function getStore() { return _store; }
