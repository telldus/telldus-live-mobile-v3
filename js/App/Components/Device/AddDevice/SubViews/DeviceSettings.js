/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

// @flow

'use strict';

import React, { useState } from 'react';
import { TextInput } from 'react-native';
import { useSelector } from 'react-redux';
import { RadioButtonInput } from 'react-native-simple-radio-button';
import { useIntl } from 'react-intl';

import {
	View,
	Text,
	CheckBoxIconText,
} from '../../../../../BaseComponents';

import Theme from '../../../../Theme';

const DeviceSettings = (props: Object): Object => {
	const {
		settings,
	} = props;

	const intl = useIntl();

	const [v, setV] = useState({});
	const [u, setU] = useState({});
	const [s, setS] = useState({});
	const [system, setSystem] = useState('');
	const [house, setHouse] = useState('');
	const [unit, setUnit] = useState('');

	const { layout } = useSelector((state: Object): Object => state.app);

	const {
		coverStyle,
		textStyle,
		radioButtonsCover,
		radioButtonLabelStyle,
		rButtonStyle,
		optionWrapStyle,
		optionButtonCover,
		buttonOuterSizeV,
		buttonSizeV,
		brandPrimary,
		uCheckBoxIconStyle,
		sRBCoverStyle,
		sRBHTextStyle,
		optionInputCover,
		optionInputLabelStyle,
		optionInputStyle,
	} = getStyles(layout);

	let Setting = [];
	Object.keys(settings).map((setting: Object) => {

		if (setting === 'system') {
			const {
				min,
				max,
			} = settings[setting];

			function onChangeText(value: string): any {

				if (!value || value === '') {
					setSystem(value);
					return;
				}

				let newValue = parseInt(value, 10);
				if (isNaN(newValue)) {
					setSystem(system);
					return;
				}

				let acceptValue = (newValue <= max) && (newValue >= min);
				if (!acceptValue) {
					setSystem(system);
					return;
				}

				setSystem(newValue.toString());
			}

			Setting.push(
				<View style={optionInputCover} key={setting}>
					<Text style={optionInputLabelStyle}>
					System
					</Text>
					<TextInput
						value={system}
						style={optionInputStyle}
						onChangeText={onChangeText}/>
				</View>
			);
		}
		if (setting === 'v') {
			const VSetting = Object.keys(settings[setting]).map((vSet: Object, index: number): Object => {
				const { option } = settings[setting][vSet];

				function onPressOne() {
					setV({
						...v,
						[vSet]: true,
					});
				}
				function onPressTwo() {
					setV({
						...v,
						[vSet]: false,
					});
				}

				const isOneSelected = v[vSet] || false;

				return (
					<View style={optionButtonCover} key={vSet}>
						<RadioButtonInput
							isSelected={isOneSelected}
							buttonStyle={rButtonStyle}
							buttonWrapStyle={optionWrapStyle}
							buttonOuterSize={buttonOuterSizeV}
							buttonSize={buttonSizeV}
							borderWidth={3}
							buttonInnerColor={brandPrimary}
							buttonOuterColor={brandPrimary}
							onPress={onPressOne}
							obj={{value: vSet}}
							index={index}/>
						<RadioButtonInput
							isSelected={!isOneSelected}
							buttonStyle={rButtonStyle}
							buttonWrapStyle={optionWrapStyle}
							buttonOuterSize={buttonOuterSizeV}
							buttonSize={buttonSizeV}
							borderWidth={3}
							buttonInnerColor={brandPrimary}
							buttonOuterColor={brandPrimary}
							onPress={onPressTwo}
							obj={{value: vSet}}
							index={index}/>
						<Text style={radioButtonLabelStyle}>
							{option}
						</Text>
					</View>
				);
			});
			Setting.push(
				<View style={radioButtonsCover} key={setting}>
					{VSetting}
				</View>);
		}
		if (setting === 'u') {
			const USetting = Object.keys(settings[setting]).map((uSet: Object, index: number): Object => {
				const { option } = settings[setting][uSet];

				const cUSet = u[uSet] || false;
				function onToggleCheckBox() {
					setU({
						...u,
						[uSet]: !cUSet,
					});
				}

				return (
					<View style={optionButtonCover} key={uSet}>
						<CheckBoxIconText
							isChecked={cUSet}
							iconStyle={{
								...optionWrapStyle,
								...uCheckBoxIconStyle,
								backgroundColor: cUSet ? brandPrimary : '#fff',
								color: cUSet ? '#fff' : 'transparent',
							}}
							textStyle={radioButtonLabelStyle}
							onToggleCheckBox={onToggleCheckBox}
							intl={intl}/>
						<Text style={radioButtonLabelStyle}>
							{option}
						</Text>
					</View>
				);
			});
			Setting.push(
				<View style={radioButtonsCover} key={setting}>
					{USetting}
				</View>);
		}
		if (setting === 's') {
			const SSetting = Object.keys(settings[setting]).map((sSet: Object, index: number): Object => {

				function onPressOne() {
					setS({
						...s,
						[sSet]: '1',
					});
				}
				function onPressTwo() {
					setS({
						...s,
						[sSet]: '-',
					});
				}
				function onPressThree() {
					setS({
						...s,
						[sSet]: '0',
					});
				}

				const isOneSelected = s[sSet] || '-';
				const isTwoSelected = s[sSet] || '-';
				const isThreeSelected = s[sSet] || '-';

				return (
					<View style={optionButtonCover} key={sSet}>
						<View style={sRBCoverStyle}>
							{index === 0 && <Text style={sRBHTextStyle}>
								{1}
							</Text>
							}
							<RadioButtonInput
								isSelected={isOneSelected === '1'}
								buttonStyle={rButtonStyle}
								buttonWrapStyle={optionWrapStyle}
								buttonOuterSize={buttonOuterSizeV}
								buttonSize={buttonSizeV}
								borderWidth={3}
								buttonInnerColor={brandPrimary}
								buttonOuterColor={brandPrimary}
								onPress={onPressOne}
								obj={{value: sSet}}
								index={index}/>
						</View>
						<View style={sRBCoverStyle}>
							{index === 0 && <Text style={sRBHTextStyle}>
								{'-'}
							</Text>
							}
							<RadioButtonInput
								isSelected={isTwoSelected === '-'}
								buttonStyle={rButtonStyle}
								buttonWrapStyle={optionWrapStyle}
								buttonOuterSize={buttonOuterSizeV}
								buttonSize={buttonSizeV}
								borderWidth={3}
								buttonInnerColor={brandPrimary}
								buttonOuterColor={brandPrimary}
								onPress={onPressTwo}
								obj={{value: sSet}}
								index={index}/>
						</View>
						<View style={sRBCoverStyle}>
							{index === 0 && <Text style={sRBHTextStyle}>
								{0}
							</Text>
							}
							<RadioButtonInput
								isSelected={isThreeSelected === '0'}
								buttonStyle={rButtonStyle}
								buttonWrapStyle={optionWrapStyle}
								buttonOuterSize={buttonOuterSizeV}
								buttonSize={buttonSizeV}
								borderWidth={3}
								buttonInnerColor={brandPrimary}
								buttonOuterColor={brandPrimary}
								onPress={onPressThree}
								obj={{value: sSet}}
								index={index}/>
						</View>
					</View>
				);
			});
			Setting.push(
				<View style={radioButtonsCover} key={setting}>
					{SSetting}
				</View>);
		}
		if (setting === 'house') {
			const {
				min,
				max,
			} = settings[setting];

			if (min || min) {
				function onChangeText(value: string): any {

					if (!value || value === '') {
						setHouse(value);
						return;
					}

					let newValue = parseInt(value, 10);
					if (isNaN(newValue)) {
						setHouse(house);
						return;
					}

					let acceptValue = (newValue <= max) && (newValue >= min);
					if (!acceptValue) {
						setHouse(house);
						return;
					}

					setHouse(newValue.toString());
				}

				Setting.push(
					<View style={optionInputCover} key={setting}>
						<Text style={optionInputLabelStyle}>
						House
						</Text>
						<TextInput
							value={house}
							style={optionInputStyle}
							onChangeText={onChangeText}/>
					</View>
				);
			}
		}
		if (setting === 'unit') {
			const {
				min,
				max,
			} = settings[setting];

			function onChangeText(value: string): any {

				if (!value || value === '') {
					setUnit(value);
					return;
				}

				let newValue = parseInt(value, 10);
				if (isNaN(newValue)) {
					setUnit(unit);
					return;
				}

				let acceptValue = (newValue <= max) && (newValue >= min);
				if (!acceptValue) {
					setUnit(unit);
					return;
				}

				setUnit(newValue.toString());
			}

			Setting.push(
				<View style={optionInputCover} key={setting}>
					<Text style={optionInputLabelStyle}>
					Unit
					</Text>
					<TextInput
						value={unit}
						style={optionInputStyle}
						onChangeText={onChangeText}/>
				</View>
			);
		}
		if (setting === 'fade') {
			Setting.push(
				<Text style={textStyle}>
					Fade
				</Text>
			);
		}
	});

	return (
		<View style={coverStyle}>
			{Setting}
		</View>
	);
};

const getStyles = (appLayout: Object): Object => {
	const { height, width } = appLayout;
	const isPortrait = height > width;
	const deviceWidth = isPortrait ? width : height;

	const {
		paddingFactor,
		shadow,
		rowTextColor,
		brandPrimary,
		eulaContentColor,
	} = Theme.Core;

	const padding = deviceWidth * paddingFactor;

	const fontSizeText = deviceWidth * 0.035;
	const fontSizeLabel = deviceWidth * 0.04;

	const outerPadding = padding * 2;

	const totalPaddingV = padding * 11;
	const buttonOuterSizeV = Math.floor((deviceWidth - (outerPadding + totalPaddingV)) / 10);
	const buttonSizeV = buttonOuterSizeV * 0.5;

	const totalPaddingU = padding * 11;
	const paddingU = 3, borderWidthU = 1;
	const iconSizeU = Math.floor((deviceWidth - (outerPadding + totalPaddingU + ((paddingU + borderWidthU) * 20))) / 10);

	return {
		paddingU,
		buttonOuterSizeV,
		buttonSizeV,
		brandPrimary,
		coverStyle: {
			marginTop: padding / 2,
			marginHorizontal: padding,
			backgroundColor: '#fff',
			...shadow,
			width: width - (2 * padding),
		},
		textStyle: {
			fontSize: fontSizeText,
			color: rowTextColor,
		},
		radioButtonsCover: {
			flexDirection: 'row',
			alignItems: 'center',
			justifyContent: 'center',
			paddingRight: padding,
			flexWrap: 'wrap',
			paddingBottom: padding,
		},
		optionButtonCover: {
			flexDirection: 'column',
			alignItems: 'center',
			justifyContent: 'center',
		},
		radioButtonLabelStyle: {
			fontSize: fontSizeText,
			color: rowTextColor,
			marginTop: padding,
		},
		rButtonStyle: {
		},
		uCheckBoxIconStyle: {
			fontSize: iconSizeU,
			borderColor: brandPrimary,
			padding: paddingU,
			borderWidth: borderWidthU,
		},
		optionWrapStyle: {
			marginTop: padding,
			marginLeft: padding,
		},
		sRBCoverStyle: {
			flexDirection: 'row',
			alignItems: 'center',
			justifyContent: 'center',
		},
		sRBHTextStyle: {
			marginTop: padding,
			marginLeft: padding,
			fontSize: fontSizeText * 1.2,
			borderColor: brandPrimary,
			textAlignVertical: 'center',
		},
		optionInputLabelStyle: {
			fontSize: fontSizeText * 1.4,
			color: rowTextColor,
		},
		optionInputCover: {
			flexDirection: 'row',
			alignItems: 'center',
			justifyContent: 'space-between',
			margin: padding,
		},
		optionInputStyle: {
			fontSize: fontSizeLabel,
			borderWidth: 1,
			borderColor: rowTextColor,
			borderRadius: 2,
			color: eulaContentColor,
			width: deviceWidth * 0.3,
			marginLeft: padding,
			padding: 5,
		},
	};
};

export default DeviceSettings;

