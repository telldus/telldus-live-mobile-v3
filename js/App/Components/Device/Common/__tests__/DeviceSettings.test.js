/**
 * Copyright 2016-present Telldus Technologies AB.
 *
 * This file is part of the Telldus Live! app.
 *
 * Telldus Live! app is free : you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Telldus Live! app is distributed in the hope this it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Telldus Live! app.  If not, see <http://www.gnu.org/licenses/>.
 *
 */


import React from 'react';
import {
	Dimensions,
} from 'react-native';
import { configureStore } from '../../../../Store/ConfigureStore';
import {act} from 'react-test-renderer';

import {
	rendererWithIntlAndRedux,
	DUMMY_DEVICE_433,
} from '../../../../../Utils/jestUtils';

import DeviceSettings from '../DeviceSettings';
import {
	getDeviceSettings,
	getDeviceVendorInfo433MHz,
} from '../../../../Lib';
import {
	setAppLayout,
	deviceInfoSuccess,
	setWidgetParamHouse,
	setWidgetParamUnit,
} from '../../../../Actions';

let {height, width} = Dimensions.get('window');

const store = configureStore();

const {
	id,
	protocol,
	model,
} = DUMMY_DEVICE_433;

let info = getDeviceVendorInfo433MHz(protocol, model);
const {
	widget,
} = info.deviceInfo;

const deviceId = id, widgetId = widget;

const setting = getDeviceSettings(parseInt(widgetId, 10), () => '');

describe('<DeviceSettings /> - snapshot', () => {
	it('renders DeviceSettings, with initializeValueFromStore', () => {
		let component;
		act(() => {
			store.dispatch(setAppLayout({
				height,
				width,
			}));

			const devices = {
				device: [
					DUMMY_DEVICE_433,
				],
			};
			store.dispatch({
				type: 'RECEIVED_DEVICES',
				payload: {
					...devices,
				},
			});
			store.dispatch(deviceInfoSuccess(DUMMY_DEVICE_433));

			component = rendererWithIntlAndRedux(
				<DeviceSettings
					settings={setting}
					deviceId={deviceId}
					widgetId={widgetId}
					initializeValueFromStore/>
			);
		});

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe('<DeviceSettings /> - snapshot', () => {
	it('renders DeviceSettings, without initializeValueFromStore', () => {
		let component;
		act(() => {
			store.dispatch(setAppLayout({
				height,
				width,
			}));

			// NOTE: Setting initial value here to prevent random selection, random selection
			// will make it impossible to use toMatchSnapshot.
			// setWidgetParamHouse, setWidgetParamUnit, is done based on DUMMY_DEVICE_433's
			// 'widget'(number). Do not update it.
			// Incase if 'protocol' or 'model' of DUMMY_DEVICE_433 is changed
			// use corresponding methods below.
			store.dispatch(setWidgetParamHouse('13'));
			store.dispatch(setWidgetParamUnit('5'));

			component = rendererWithIntlAndRedux(
				<DeviceSettings
					settings={setting}/>
			);
		});

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
