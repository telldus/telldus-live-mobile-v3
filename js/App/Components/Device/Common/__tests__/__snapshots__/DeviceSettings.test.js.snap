// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeviceSettings /> - snapshot renders DeviceSettings, with initializeValueFromStore 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "marginHorizontal": 20.83275,
        "marginTop": 10.416375,
        "width": 708.3345,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#777777",
        "fontFamily": "Roboto-Regular",
        "fontSize": 33,
        "marginBottom": 5,
        "marginTop": 10.416375,
      }
    }
  >
    Device settings
  </Text>
  <View
    accessibilityLabel="House code, 12 , "
    accessible={true}
    importantForAccessibility="yes"
    style={
      Array [
        Object {
          "backgroundColor": "#f5f5f5",
          "borderRadius": 2,
          "elevation": 2,
          "marginTop": 10.416375,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2,
        },
        Object {
          "borderRadius": 2,
          "marginBottom": 10.416375,
          "marginTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 2,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "overflow": "hidden",
            "padding": 30,
          },
          Object {
            "padding": 20.83275,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#000",
                "fontSize": 30,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          House code
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 2,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            importantForAccessibility="yes"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#9C7D69",
                "fontFamily": "telldusicons",
                "fontSize": 30,
              }
            }
          >
            help
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "opacity": 1,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#4C4C4C",
                "flex": 1,
                "fontSize": 30,
                "textAlign": "right",
                "textAlignVertical": "center",
              },
              Object {
                "flex": 0,
              },
            ]
          }
        >
          12
           
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": 3,
              "opacity": 1,
              "padding": 2,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Óèâ
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="Unit code, 4 , "
    accessible={true}
    importantForAccessibility="yes"
    style={
      Array [
        Object {
          "backgroundColor": "#f5f5f5",
          "borderRadius": 2,
          "elevation": 2,
          "marginTop": 10.416375,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2,
        },
        Object {
          "borderRadius": 2,
          "marginBottom": 10.416375,
          "marginTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 2,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "overflow": "hidden",
            "padding": 30,
          },
          Object {
            "padding": 20.83275,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#000",
                "fontSize": 30,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          Unit code
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 2,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            importantForAccessibility="yes"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#9C7D69",
                "fontFamily": "telldusicons",
                "fontSize": 30,
              }
            }
          >
            help
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "opacity": 1,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#4C4C4C",
                "flex": 1,
                "fontSize": 30,
                "textAlign": "right",
                "textAlignVertical": "center",
              },
              Object {
                "flex": 0,
              },
            ]
          }
        >
          4
           
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": 3,
              "opacity": 1,
              "padding": 2,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Óèâ
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<DeviceSettings /> - snapshot renders DeviceSettings, without initializeValueFromStore 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "marginHorizontal": 20.83275,
        "marginTop": 10.416375,
        "width": 708.3345,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#777777",
        "fontFamily": "Roboto-Regular",
        "fontSize": 33,
        "marginBottom": 5,
        "marginTop": 10.416375,
      }
    }
  >
    Device settings
  </Text>
  <View
    accessibilityLabel="House code, 13 , "
    accessible={true}
    importantForAccessibility="yes"
    style={
      Array [
        Object {
          "backgroundColor": "#f5f5f5",
          "borderRadius": 2,
          "elevation": 2,
          "marginTop": 10.416375,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2,
        },
        Object {
          "borderRadius": 2,
          "marginBottom": 10.416375,
          "marginTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 2,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "overflow": "hidden",
            "padding": 30,
          },
          Object {
            "padding": 20.83275,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#000",
                "fontSize": 30,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          House code
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 2,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            importantForAccessibility="yes"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#9C7D69",
                "fontFamily": "telldusicons",
                "fontSize": 30,
              }
            }
          >
            help
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "opacity": 1,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#4C4C4C",
                "flex": 1,
                "fontSize": 30,
                "textAlign": "right",
                "textAlignVertical": "center",
              },
              Object {
                "flex": 0,
              },
            ]
          }
        >
          13
           
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": 3,
              "opacity": 1,
              "padding": 2,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Óèâ
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="Unit code, 5 , "
    accessible={true}
    importantForAccessibility="yes"
    style={
      Array [
        Object {
          "backgroundColor": "#f5f5f5",
          "borderRadius": 2,
          "elevation": 2,
          "marginTop": 10.416375,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.23,
          "shadowRadius": 2,
        },
        Object {
          "borderRadius": 2,
          "marginBottom": 10.416375,
          "marginTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 2,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "overflow": "hidden",
            "padding": 30,
          },
          Object {
            "padding": 20.83275,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#000",
                "fontSize": 30,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          Unit code
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 2,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            importantForAccessibility="yes"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#9C7D69",
                "fontFamily": "telldusicons",
                "fontSize": 30,
              }
            }
          >
            help
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "opacity": 1,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#4C4C4C",
                "flex": 1,
                "fontSize": 30,
                "textAlign": "right",
                "textAlignVertical": "center",
              },
              Object {
                "flex": 0,
              },
            ]
          }
        >
          5
           
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": 3,
              "opacity": 1,
              "padding": 2,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 30,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Óèâ
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
